### PR TITLE
Update Cache Config To Use TimeSpan

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp.Web/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Updates configuration to allow caching by any `TimeSpan` configuration. See #104 

This is a necessary breaking change.

I've also reintroduced the samples webapp into the solution.

### CODE REVIEW

**Review by tocsoft — COMMENTED** [review]
I do think I would just make more sense to just have a single option called `MaxCacheTimeout` and it be of type `TimeSpan` instead of the now 3 separate parts it then allows for super fine grained adjustments without have to keep adding additional properties and having to then coherence the multiple files into a `TimeSpan` anyway.

**Comment by JimBobSquarePants** [comment]
Yeah, we'd have to determine a format though that can be parsed invariantly.

**Comment by tocsoft** [comment]
`TimeSpan`s just work for serializing/parsing, if we are loading from `app.config`s anyway.

**Comment by tocsoft** [comment]
https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-timespan-format-strings

The invariant format looks like this `3.17:25:30.5000000` `[d'.']hh':'mm':'ss['.'fffffff]`  so generally HH:MM:SS with optional days prefix and milliseconds postfix

**Comment by JimBobSquarePants** [comment]
> TimeSpans just work for serializing/parsing, if we are loading from app.configs anyway.

Beautiful. I'll make the changes.

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -74,7 +74,7 @@ jobs:
                   XUNIT_PATH: .\tests\ImageSharp.Web.Tests # Required for xunit
 
             - name: Update Codecov
-              uses: codecov/codecov-action@v1.0.7
+              uses: codecov/codecov-action@v1
```
Comment (JimBobSquarePants) [inline .github/workflows/build-and-test.yml:77]: We'll have to do this in all our repos. They've changed things which causes a 400

**Review by tocsoft — APPROVED** [review]
much better 👍

**Pull request merged**